slither/detectors/naming_convention/naming_convention.py

```diff
@@ -21,6 +25,12 @@ def is_cap_words(name):
     def is_mixed_case(name):
         return re.search('^[a-z]([A-Za-z0-9]+)?_?$', name) is not None
 
+    @staticmethod
+    def is_mixed_case_with_underscore(name):
+        # Allow _ at the beginning to represent private variable
+        # or unused parameters
+        return re.search('^[a-z_]([A-Za-z0-9]+)?_?$', name) is not None
```

Comment:
```suggestion
        return re.search('^[a-z_][A-Za-z0-9]*_?$', name) is not None
```
Also we should compile the regex `SOME_REGEX = re.compile(...)` and use it here.